### ISSUE TITLE
Move the oc tool to a collect-sysinfo option

### ISSUE THREAD

**Issue by portante** [distributed-system-analysis/pbench#552]
As far as we can tell, the current `oc` tool appears to be collecting system information rather than metrics.

The tool collects a series of data at the start and at the end of each tool invocation, and nothing periodically in between.  Here is a by-hand expansion of the oc commands performed as a result of the typical tool script invocation:

1. `tool-scripts/oc --start`
    ```
    sleep 5
    oc get --all-namespaces rc -o wide -w | unbuffer -p awk '{print strftime("%Y-%m-%d %H:%M:%S"),$0}' >> $tool_output_dir/rc.txt &
    sleep 5
    oc get --all-namespaces ep -o wide -w | unbuffer -p awk '{print strftime("%Y-%m-%d %H:%M:%S"),$0}' >> $tool_output_dir/ep.txt &
    sleep 5
    oc get --all-namespaces pods -o wide -w | unbuffer -p awk '{print strftime("%Y-%m-%d %H:%M:%S"),$0}' >> $tool_output_dir/pods.txt &
    sleep 5
    oc get pv -o wide -w | unbuffer -p awk '{print strftime("%Y-%m-%d %H:%M:%S"),$0}' >> $tool_output_dir/pv.txt &
    sleep 5
    oc get --all-namespaces pvc -o wide -w | unbuffer -p awk '{print strftime("%Y-%m-%d %H:%M:%S"),$0}' >> $tool_output_dir/pvc.txt &
    sleep 5
    oc get --all-namespaces svc -o wide -w | unbuffer -p awk '{print strftime("%Y-%m-%d %H:%M:%S"),$0}' >> $tool_output_dir/svc.txt &
    sleep 5
    oc get cs -o wide -w | unbuffer -p awk '{print strftime("%Y-%m-%d %H:%M:%S"),$0}' >> $tool_output_dir/cs.txt &
    ```
1. `tool-scripts/oc --stop`
    ```
    oc get nodes -o wide >> "${tool_output_dir}"/nodes.txt 
    oc get ev --all-namespaces -o wide >> "${tool_output_dir}"/ev.txt 
    ```
1. `tool-scripts/oc --postprocess`
    ```
    oc get nodes -o wide >> "${tool_output_dir}"/nodes.txt 
    oc get ev --all-namespaces -o wide >> "${tool_output_dir}"/ev.txt 
    ```

First of all, none of this appears to be tool data of collected metrics in a timeseries.  This is all configuration data that is collected once.  This has the markings of "sysinfo" data, and likely should be moved to that facility, and removed as a tool.

However, that would mean we would not collect this information at the start of every iteration.

Note, too, that after tool "stop" operation comes a "postprocess" operation, so having the nodes and ev collected in close proximity seems a waste.

Finally, all of the data collection for the start tool is performed in the background, but the sleeps are not.  So its takes a full 35 seconds before a test can proceed.  Additionally, if any of those command take longer than 5 seconds, we'll add another process invoking the APIs, which might cause further harm.  Seems like we need to "wait" after we background the oc command, and only proceed if 5 or more seconds have elapsed before issuing the next oc command.

@jeremyeder, please consider and review.

**Comment by jeremyeder** [comment]
Doesn't collect-sysinfo run just once at the beginning and once again at the end?

The idea behind this tool is to answer how many X objects exist at intervals throughout a test run.

**Comment by portante** [comment]
It does not appear to do that.

It only runs one before a test and the collects other data at the end.

**Comment by jeremyeder** [comment]
I'm not actually sure anything's wrong yet. @ekuric can you please peel back what's going on here? 

The -w option to oc establishes a client watch against etcd for that resource.  The watch never ends, and that is by design.

Without doing a watch, it is too abusive on the apiserver when we are at scale.  We encountered this years ago and it's still true.  Watches are used aggressively everywhere in kube for this reason (in fact it's the main characteristic that made kube choose etcd in the first place).

The problem with watch is that it only prints changes as the watch is internally notified of them, which is an importantly different behavior than the UNIX "watch" command.

**Comment by portante** [comment]
@jeremyeder, thanks that makes sense.  I can see the `-w` option, and had not known what the option does.  At the least, I think we need to modify this code to add comments to explain the behavior here, as it is not intuitive, and as this ages, won't be well supported.

Yet, even with that explanation, this still does not appear to be a typical pbench tool.

Even though we have timestamps captured for the "watched" data, we don't have any of that data presented beyond what is dumped into the text files.

The data could be collected in the same manner, just done via collect-sysinfo hook points.

The `get nodes` and `get ev` are not performed as watches, and are not timestamped, with the postprocess invocation really a duplicate of the data captured via stop, and it is different data by nature from the "watched" data.

And the 35 second startup time seems like arbitrary overhead.

The one exception to this argument, is that for benchmarks like uperf and fio, each iteration has all this data collected (start, stop, postprocess) as separate files, so it makes it easy to see what data is related to what iteration of the test.  Given the data is timestamped, I think we can correlate that data later, though, especially since we are going to index it (I would assume).

**Comment by ekuric** [comment]
@portante oc waits on event to happen and then collect that info, and not only at test start/end ( -w switch is responsible for that)

**Comment by portante** [comment]
@ekuric, see my previous comment, too.

**Comment by ashishkamra** [comment]
need team discussion

**Comment by portante** [comment]
Given this request is almost 5 years old, we are not going to try to address this enhancement.  If somebody feels that this is in error, please re-open the issue with additional feedback describing the need for changes.
